tests/components/lovelace/test_dashboard.py

```diff
@@ -275,3 +293,154 @@
+    response = await client.receive_json()
+    assert response["success"]
+    assert response["result"]["require_admin"] is True
+    assert response["result"]["sidebar"] == {
+        "title": "Updated Title",
+        "icon": "mdi:map",
+    }
+
+    dashboard_id = response["result"]["id"]
+
+    assert "created_url_path" in hass.data[frontend.DATA_PANELS]
+
+    await client.send_json({"id": 7, "type": "lovelace/dashboards/list"})
+    response = await client.receive_json()
+    assert response["success"]
+    assert len(response["result"]) == 1
+    assert response["result"][0]["mode"] == "storage"
+    assert response["result"][0]["sidebar"] == {
+        "title": "Updated Title",
+        "icon": "mdi:map",
+    }
+    assert response["result"][0]["require_admin"] is True
+
+    # Fetch config
+    await client.send_json(
+        {"id": 8, "type": "lovelace/config", "url_path": "created_url_path"}
+    )
+    response = await client.receive_json()
+    assert not response["success"]
+    assert response["error"]["code"] == "config_not_found"
+
+    # Store new config
+    events = async_capture_events(hass, const.EVENT_LOVELACE_UPDATED)
+
+    await client.send_json(
+        {
+            "id": 9,
+            "type": "lovelace/config/save",
+            "url_path": "created_url_path",
+            "config": {"yo": "hello"},
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    assert hass_storage[dashboard.CONFIG_STORAGE_KEY.format(dashboard_id)]["data"] == {
+        "config": {"yo": "hello"}
+    }
+    assert len(events) == 1
+    assert events[0].data["url_path"] == "created_url_path"
+
+    await client.send_json(
+        {"id": 10, "type": "lovelace/config", "url_path": "created_url_path"}
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    assert response["result"] == {"yo": "hello"}
+
+    # Update a dashboard
+    await client.send_json(
+        {
+            "id": 11,
+            "type": "lovelace/dashboards/update",
+            "dashboard_id": dashboard_id,
+            "require_admin": False,
+            "sidebar": None,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    assert response["result"]["require_admin"] is False
+    assert "sidebar" not in response["result"]
+
+    # Add dashboard with existing url path
+    await client.send_json(
+        {"id": 12, "type": "lovelace/dashboards/create", "url_path": "created_url_path"}
+    )
+    response = await client.receive_json()
+    assert not response["success"]
+
+    # Delete dashboards
+    await client.send_json(
+        {"id": 13, "type": "lovelace/dashboards/delete", "dashboard_id": dashboard_id}
+    )
+    response = await client.receive_json()
+    assert response["success"]
+
+    assert "created_url_path" not in hass.data[frontend.DATA_PANELS]
+    assert dashboard.CONFIG_STORAGE_KEY.format(dashboard_id) not in hass_storage
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5093948)